src/app/core/feedback/feedback.service.js

```diff
@@ -56,16 +70,51 @@ const search = async (reqUser, queryParams, query) => {
 	const offset = page * limit;
 
 	// Query for feedback
-	const feedback = await Feedback.textSearch(query, null, limit, offset, sortArr, true, {
+	const feedback = await Feedback.textSearch(
+		query,
+		null,
+		limit,
+		offset,
+		sortArr,
+		true,
+		{
+			path: 'creator',
+			select: ['username', 'organization', 'name', 'email']
+		}
+	);
+
+	return util.getPagingResults(limit, page, feedback.count, feedback.results);
+};
+
+const getFeedbackById = async (feedbackId) => {
+	return Feedback.findOne({
+		_id: mongoose.Types.ObjectId(feedbackId)
+	}).populate({
 		path: 'creator',
-		select: ['name', 'email']
+		select: ['username', 'organization', 'name', 'email']
 	});
+};
 
-	return util.getPagingResults(limit, page, feedback.count, feedback.results);
+const updateFeedbackAssignee = async (feedbackId, assignee) => {
+	const feedback = await getFeedbackById(feedbackId);
+	feedback.assignee = assignee;
+	feedback.updated = Date.now();
+	await feedback.save();
+	return feedback;
+};
+
+const updateFeedbackStatus = async (feedbackId, status) => {
+	const feedback = await getFeedbackById(feedbackId);
```

Comment:
There is no error handling for case where an invalid feedbackId is passed in.  I did some quick testing and result was an internal server.

Ideally we should use an express param handler to abstract this out from the service.  In that case the service is guaranteed to receive a valid feedback.

We already use express param handlers in multiple controllers.  Here are examples of how we do this for teams.
https://github.com/Asymmetrik/node-rest-starter/blob/develop/src/app/core/teams/teams.routes.js#L49
https://github.com/Asymmetrik/node-rest-starter/blob/develop/src/app/core/teams/teams.controller.js#L249

Reply:
@jrassa This is quite elegant. I'll implement it for the feedback!

Reply:
- [ ] Update feedbackId param handling

Reply:
@jrassa Hi John. I've updated this parameter's validation as per your suggestions here and our slack discussion. Just to make sure I did everything as expected, the changes specifically for this are:
- Addition of `router.param('feedbackId', feedback.feedbackById);` in the `feedback.routes.js` file
- Addition of `feedbackById` in the `feedback.controller.js` file. Here, I've wrapped the service call to `readFeedback` in a try/catch block to handle the expected and unexpected errors.
- In the `readFeedback` method in the `feedback.service.js` file, I've inserted 2 checks that will potentially throw errors, one for the validity of the ID, and the other for if feedback exists or not.

Does all of that look correct?

Reply:
@jrassa Hi John. I've updated this parameter's validation as per your suggestions here and our slack discussion. Just to make sure I did everything as expected, the changes specifically for this are:

- Addition of `router.param('feedbackId', feedback.feedbackById);` in the `feedback.routes.js` file
- Addition of `feedbackById` in the `feedback.controller.js` file. Here, I've wrapped the service call to `readFeedback` in a try/catch block to handle the expected and unexpected errors.
In the `readFeedback` method in the `feedback.service.js` file, I've inserted 2 checks that will potentially throw errors, one for the validity of the ID, and the other for if feedback exists or not.

Does all of that look correct?

Reply:
Looks good.  I would have maybe had the null check in the controller vs. the service in case we want to reuse the `readFeedback` method elsewhere where we might not want/need to throw an error.  Since it isn't actually used anywhere else, it is fine as is.  It can always be updated later if we have the need.

Reply:
I'll go ahead and move the null check up into the controller! Better to do things right up front :)

- [ ] Move null check into controller